Feat(Dashboard): Revamp layout

usame-algan opened the pull request:
## What it solves

Resolves [EN-127](https://linear.app/safe-global/issue/EN-127/dashboard-update-header)

## How this PR fixes it

- Adds a new `NewsCarousel` component
- Creates new banners for spaces and earn and adds them to the carousel
- Updates the layout of the Overview component according to Figma

## How to test it

1. Open a Safe
2. Observe the new Dashboard overview

## Screenshots

With banners visible:
<img width="1266" alt="Screenshot 2025-06-10 at 10 55 28" src="https://github.com/user-attachments/assets/cec0c111-c5ec-4541-ad06-6a45e72709bc" />

With all banners dismissed:
<img width="1283" alt="Screenshot 2025-06-10 at 10 53 07" src="https://github.com/user-attachments/assets/aaa8ec9e-8d2d-403d-84f8-556b08d5627e" />

Mobile:
<img width="422" alt="Screenshot 2025-06-10 at 11 56 59" src="https://github.com/user-attachments/assets/82529561-310e-4e18-86e3-deac8e979b31" />

Dark mode:
<img width="1269" alt="Screenshot 2025-06-10 at 11 57 12" src="https://github.com/user-attachments/assets/eb904737-6e60-4c49-b426-4b02c63b97e8" />

## Checklist

- [ ] I've tested the branch on mobile 📱
- [ ] I've documented how it affects the analytics (if at all) 📊
- [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

---

## CLA signature

With the submission of this Pull Request, I confirm that I have read and agree to the terms of the [Contributor License Agreement](https://safe.global/cla).

compojoom commented:
@usame-algan @TanyaEfremova I just tested it and it feels a bit werid. 

When you have 2 banners you have this shadow on the right indicating that you can scroll there by clicking on this erea. It really scrolls to the second banner, but I cannot click on the left to go back to the first banner. I have to click on the 2 dots at the bottom. I don't know this somehow screws with my brain. It's weird.
![grafik](https://github.com/user-attachments/assets/0677e9ac-bb37-46e4-868a-85808da9c454)
and then:
![grafik](https://github.com/user-attachments/assets/c300ecae-800d-43ef-bc42-a608b0bdfe56)


to me this looks broken. I would immediately wonder - why is this not stretching.

usame-algan commented:
> I would immediately wonder - why is this not stretching.

We could stretch it if its only one item. @TanyaEfremova wdyt?

I wanted to start with as little as possible for the carousel. We can add more interactions if necessary.

TanyaEfremova commented:
Additionally, swiping should be possible by holding and dragging the mouse, it's hard to click on the dot below. And/or we could add a chevron on the carousel sides, for even better accessibility.

usame-algan commented:
@TanyaEfremova I've added a swipe interaction. Not sure if the dots are even needed now since the next banner is always visible to indicate to the user that there is more. I also removed the overlay on the right because it was overlapping the close button of the second banner. Not sure how to solve that. Do you think it would work without?

francovenica commented:
The only issue I had is that sometimes I grab the image when I try to scroll the card:
![drag card](https://github.com/user-attachments/assets/a0f94fe5-4c82-4055-b990-c6f780d15655)

Beyond that the feature looks fine to me
